lib/tasks/generate/runMachine.js

```diff
@@ -0,0 +1,116 @@
+var { spawn }     = require('child-process-promise');
+var npmInstall    = require('enpeem').install;
+var fs            = require('fs-extra');
+var path          = require('path');
+
+var dockerCompose = require('../../helpers/dockerCompose')();
+
+
+module.exports = {
+  /**
+   * @overview This function executes the machine that is responsible for running client tasks
+   *
+   * @param {String} blueprintPath - path to the blueprint/addon
+   * @param {Object} [options] - Options passed over by commander including all set flags such --verbose
+   * @return {Promise} promise - So we know when everything has been executed.
+  */
+  client: function (blueprintPath, projectRoot, clientName, options) {
+    var blueprint = require(blueprintPath);
+
+    var promise = new Promise(function (resolve) { //reject
+      blueprint.client.exec({
+        success: async function (actions) {
+          //only install ember-addons addons if --skip-npm is not set
+          if (!options.skipNpm) {
+            for (var emberAddon of actions.addEmberAddons) {
+              var addonSemVer = emberAddon.name + '@' + emberAddon.target;
+              var emberArgs = ['install', addonSemVer];
+              try {
+                await spawn('ember', emberArgs, { stdio: 'inherit', cwd: path.join(projectRoot, clientName) });
```

Comment:
Need to use ember helper here, not hardcoded `'ember'`.  Again on line 40 as well.